voxel51/platform/task.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, d):
             d, "inputs", voxu.RemotePathConfig, default={})
         self.parameters = self.parse_dict(d, "parameters", default={})
         self.status = self.parse_object(d, "status", voxu.RemotePathConfig)
-        self.logfile = self.parse_object(d, "logfile", voxu.RemotePathConfig)
+        self.log = self.parse_object(d, "logfile", voxu.RemotePathConfig)
```

Comment:
This breaks backwards compatibility for old analytics running on new versions of the Platform. Wasn't the point to preserve that?

Reply:
I don't see how this breaks compatibility - old images won't change, and the underlying JSON structure is the same. This field name is only used in upload_logfile, so unless users only update _part_ of the SDK, this should continue to work. I suppose user code could access this field directly, though.

Reply:
Oh I see, you didn't change `"logfile"` on the left, you just changed the attribute name. That's an antipattern for `eta.core.config.Config`, which, by convention, expects that you can do `Config.from_dict(config.serialize())`. Your name changing would break that.

If you really want to access this field via `log`, then just add a property to the class that aliases `logfile` to `log`. Note, `property`, not just another attribute, because if you added another attribute, that would include the extra field in the serialized dict generated by `TaskConfig.serialize()`. Just an ETA style thing.

Reply:
Ordinarily this wouldn't be re-serializing the TaskConfig, but I'll change it back for consistency's sake. (I'm not sure this change was intentional anyway... this field was removed entirely in 5718110bcdafd39d4b3a44dd56ba253bea7e39eb, and later re-added in c6216b56d776aef9122cfb2069af66c84fbe3cdc)

Reply:
Oh, this was changed because the API uses `log` in the route:
https://github.com/voxel51/platform-sdk/pull/25/files#diff-e59463f9e80523327f57336a85490760R127
https://github.com/voxel51/platform-sdk/pull/25/files#diff-e59463f9e80523327f57336a85490760R216

Reply:
Right, you changed it because it made the rest of your code cleaner, but it should be `TaskConfig.logfile` for best `Config` practices

Reply:
I stopped short of saying the API route should use `logfile`, although I thought about it ;-)

Reply:
Looked into changing it in a few other places, but I think the alias you suggested is the simplest.